File: C++/second_largest.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;
    vector<int> nums(n);
    for (int i=0; i<n; i++){
        cin >> nums[i];
    }
    int largest = nums[0];
    int slargest = -1;
    for(int i : nums){
        if(i > largest){
            slargest = largest;
            largest = i;
        }
        else if(i < largest && i > slargest){
            slargest = i;
        }
    }
    cout << slargest << endl;
}

File: Python/second_largest.py
n = int(input())
nums = []
for _ in input().split():
    nums.append(int(_))

largest = nums[0]
slargest = -1
for i in nums:
    if i > largest:
        slargest = largest
        largest = i
    elif i > slargest and i != largest:
        slargest = i
print(slargest)